lib/cufft/fft.jl

```diff
@@ -34,14 +34,16 @@ mutable struct CuFFTPlan{T<:cufftNumber,S<:cufftNumber,K,inplace,N} <: Plan{S}
     input_size::NTuple{N,Int}   # Julia size of input array
     output_size::NTuple{N,Int}  # Julia size of output array
     region::Any
+    buffer::CuArray{S}
     pinv::ScaledPlan{T}         # required by AbstractFFTs API, will be defined by AbstractFFTs if needed
 
     function CuFFTPlan{T,S,K,inplace,N}(handle::cufftHandle,
                                         input_size::NTuple{N,Int}, output_size::NTuple{N,Int}, region
                                         ) where {T<:cufftNumber,S<:cufftNumber,K,inplace,N}
         abs(K) == 1 || throw(ArgumentError("FFT direction must be either -1 (forward) or +1 (inverse)"))
         inplace isa Bool || throw(ArgumentError("FFT inplace argument must be a Bool"))
-        p = new{T,S,K,inplace,N}(handle, context(), stream(), input_size, output_size, region)
+        buffer = CuArray{S}(0)
```

Comment:
I wonder if we shouldn't be using `nothing` instead. That should flush out an accidentally-used buffer earlier, rather than a more inscrutable BoundsError when accidentally using the buffer.